kong/clustering/rpc/utils.lua

```diff
@@ -13,13 +14,26 @@ local snappy_compress = snappy.compress
 local snappy_uncompress = snappy.uncompress
 
 
+local cap_names = lrucache.new(100)
+
+
 function _M.parse_method_name(method)
+  local cap = cap_names:get(method)
```

Comment:
The method should not very long, I'm wondering on if the lrucache is significant faster than this simple string operation.

I'm trying to avoid meaningless memory consumption.

Reply:
Yes, I also notice that, need more discussion.

Reply:
I did a simple perf test:
```
-- lru cache
local function parse_method_name1(method) ... end

-- no cache
local function parse_method_name2(method) ... end

local COUNT = 100 * 1000

ngx.update_time()
local t = ngx.now()

for i = 1, COUNT do
  parse_method_name1("kong.rpc.v1.test")
end

ngx.update_time()
print("v1: ", ngx.now() - t)

local t = ngx.now()

for i = 1, COUNT do
  parse_method_name2("kong.rpc.v1.test")
end

ngx.update_time()
print("v2: ", ngx.now() - t)
```
The result is：
```
resty ./t.lua
v1: 0.002000093460083
v2: 0.0079998970031738
```
So cache is faster than calculation.

Reply:
This test is on the happy path of the scenario, so it's not convincing enough. Please consider that we may have a different ratio of cache hit; also consider the balance between memory and CPU time.

Please consider using tables to cache directly. I cannot think of a reason to use LRU.

Reply:
@StarlightIbuki The RPC protocol should have limited methods, and the hit rate should be very high if the capacity makes sense.

And all methods must be registered on-demand, we don't have a global method list, it depends on the peer of RPC. 

To eliminate the LRU and string operations, we have to maintain a Lua table for each connected RPC peer, it tends to consume more memory.

@chronolaw Please correct me if anything is wrong.

Reply:
It may be a concern, I think that you could open a new PR then we can discuss in it.